common/src/main/java/com/wynntils/services/itemfilter/statproviders/PriceStatProvider.java

```diff
@@ -4,18 +4,27 @@
  */
 package com.wynntils.services.itemfilter.statproviders;
 
+import com.wynntils.core.components.Models;
 import com.wynntils.models.items.WynnItem;
-import com.wynntils.models.items.items.game.GameItem;
+import com.wynntils.models.items.WynnItemData;
+import com.wynntils.models.trademarket.type.TradeMarketPriceInfo;
 import com.wynntils.services.itemfilter.type.ItemStatProvider;
 import java.util.List;
 
 public class PriceStatProvider extends ItemStatProvider<Integer> {
     @Override
     public List<Integer> getValue(WynnItem wynnItem) {
-        if (wynnItem instanceof GameItem gameItem && gameItem.hasEmeraldPrice()) {
-            return List.of(gameItem.getEmeraldPrice());
+        TradeMarketPriceInfo priceInfo = wynnItem.getData().getOrCalculate(WynnItemData.EMERALD_PRICE_KEY, () -> {
```

Comment:
I don't think you need the block around the lambda here..?

Reply:
You do, since you need to get the itemstack.

Reply:
I don't get it. Why can't you just do `() -> Models.TradeMarket.calculateItemPriceInfo(wynnItem.getData().get(WynnItemData.ITEMSTACK_KEY))`?

Reply:
Oh, yes, I can. But it arguably looks worse with spotless...

![image](https://github.com/Wynntils/Artemis/assets/49001742/ee0c38f7-cb07-4cf8-b55b-89c6de56db4d)

Reply:
Ok, fine, then it's just a matter of taste. I think the code above looks better but I accept that you prefer the current version. As long as the reason for the current code was not "oh, i did more stuff before, but I refactored and forget to simplify".